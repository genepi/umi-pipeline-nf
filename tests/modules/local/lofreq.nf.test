nextflow_process {

    name "Test Process LOFREQ"
    script "modules/local/variant_calling/lofreq.nf"
    process "LOFREQ"

    test("Should run without failures") {

        when {
            params {
                variant_calling = true 
                variant_caller = "lofreq"
            }
            process {
                """
                input[0] = Channel.of(["barcode03", "lpa2645", "$baseDir/tests/input/variant_calling/barcode03/final.bam", "$baseDir/tests/input/variant_calling/barcode03/final.bam.bai"])
                input[1] = "final"
                input[2] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta")
                input[3] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta.fai")
                """
            }
        }

        then {
            assert process.success
            with(process.out.variants){
                assert path(get(0).get(1)).vcf.summary == path("$baseDir/tests/validation/lofreq/${get(0).get(0)}/final.vcf").vcf.summary
            }
        }
    }

    test("Should output three files") {

        when {
            params {
                variant_calling = true 
                variant_caller = "lofreq"
            }
            process {
                """
                input[0] = Channel.of(
                    ["barcode03", "lpa2645", "$baseDir/tests/input/variant_calling/barcode03/final.bam", "$baseDir/tests/input/variant_calling/barcode03/final.bam.bai"],
                    ["barcode04", "lpa2645", "$baseDir/tests/input/variant_calling/barcode04/final.bam", "$baseDir/tests/input/variant_calling/barcode04/final.bam.bai"],
                    ["barcode05", "lpa2645", "$baseDir/tests/input/variant_calling/barcode05/final.bam", "$baseDir/tests/input/variant_calling/barcode05/final.bam.bai"]
                    )
                input[1] = "final"
                input[2] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta")
                input[3] = file("$baseDir/tests/input/pipeline/ref/lpa-ref2645.fasta.fai")
                """
            }
        }

        then {
            assert process.success
            with(process.out.variants){
                assert path(get(0).get(1)).vcf.summary == path("$baseDir/tests/validation/lofreq/${get(0).get(0)}/final.vcf").vcf.summary
                assert path(get(1).get(1)).vcf.summary == path("$baseDir/tests/validation/lofreq/${get(1).get(0)}/final.vcf").vcf.summary
                assert path(get(2).get(1)).vcf.summary == path("$baseDir/tests/validation/lofreq/${get(2).get(0)}/final.vcf").vcf.summary
            }
        }

    }

}
