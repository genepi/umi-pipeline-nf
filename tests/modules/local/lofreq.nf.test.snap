{
    "Should output three files": {
        "content": [
            {
                "0": [
                    "final.vcf:md5,22906ceda834065e1d62878ed521f18c",
                    "final.vcf:md5,9d0d55c36e3f6f72b65371b3ea534260",
                    "final.vcf:md5,9da201124f97ba4ffcea42720219ce88"
                ],
                "variants": [
                    "final.vcf:md5,22906ceda834065e1d62878ed521f18c",
                    "final.vcf:md5,9d0d55c36e3f6f72b65371b3ea534260",
                    "final.vcf:md5,9da201124f97ba4ffcea42720219ce88"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "22.10.3"
        },
        "timestamp": "2024-09-13T15:43:14.912982"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "final.vcf:md5,9d0d55c36e3f6f72b65371b3ea534260"
                ],
                "variants": [
                    "final.vcf:md5,9d0d55c36e3f6f72b65371b3ea534260"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "22.10.3"
        },
        "timestamp": "2024-09-13T15:43:10.490675"
    }
}